graphql_compiler/compiler/compiler_frontend.py

```diff
@@ -447,10 +461,12 @@ def _compile_vertex_ast(schema, current_schema_type, ast,
         if not fold_directive and not is_in_fold_scope(context):
             # Current block is either a Traverse or a Recurse that is not within any fold context.
             # Increment the `num_traverses` counter.
-            current_location_stack_entry = context['marked_location_stack'][-1]
-            current_num_traverses = current_location_stack_entry.num_traverses
-            new_location_stack_entry = current_location_stack_entry._replace(
-                num_traverses=current_num_traverses+1)
+            old_location_stack_entry = context['marked_location_stack'][-1]
+            old_location = old_location_stack_entry.location
+            old_num_traverses = old_location_stack_entry.num_traverses
+
+            new_location_stack_entry = _construct_location_stack_entry(
+                old_location, old_num_traverses+1)
```

Comment:
You'll need spaces around the `+`, and ideally use `old_location_stack_entry.location` and `old_location_stack_entry.num_traverses + 1` directly here